src/map/instance.cpp

```diff
@@ -257,3 +259,130 @@ bool CInstance::CheckFirstEntry(uint32 id)
     //insert returns a pair (iterator,inserted)
     return m_enteredChars.insert(id).second;
 }
+
+void CInstance::DisengageAll()
```

Comment:
I was using it in most of them, but I'm not sure its needed now. I had to do this manually before i fixed the event issues below. Now I notice some of the events cause this to happen as a side effect. if that's the case I may be able to drop this entirely. I had already started removing it where it wasn't strictly necessary. As for doing it in a script, there are no bindings to directly engage or disengage a mob.